include/translate-option.php

```diff
@@ -130,13 +130,16 @@ public function translate( $value ) {
 	 *
 	 * @param mixed      $values Either a string to translate or a list of strings to translate.
 	 * @param array|bool $key    Array of option keys to translate.
-	 * @return array|string Translated string(s)
+	 * @return array|string Translated string(s).
 	 */
 	protected function translate_string_recursive( $values, $key ) {
 		$children = is_array( $key ) ? $key : array();
 
 		if ( is_array( $values ) || is_object( $values ) ) {
+			/** @var array|Traversable $values */
```

Comment:
Could we pass only `array` to the matcher using `iterator_to_array()`? And thus avoid to manage several type in `PLL_Format_Utils`?

Reply:
Weird, I can't find any use of `filter_list()` anymore where the 1st argument is not an array, neither in PLL and PLL Pro, while I remember we had a case with `stdClass`. I guess it disappeared in a PR.
Anyway, done in 77a0d42.

Now I wonder if we should tag this class and methods `@since 3.7` or `@since 3.6` + a `@since 3.7 Changed signature` for `filter_list()` 🤔

Reply:
Could it me some third party plugin that stores its options to translate in an object?

Reply:
Each time we use `filter_list()`, we make sure to give it an array: `is_array()` x3 in PLL Pro, `(array)` cast x1 in PLL (master).

Reply:
Sorry, I missed your last question.
For the tags I'd say; `@since3.6` + `@since 3.7 Moved from Pro to Free`. What do you think?

Reply:
Done in [080af71](https://github.com/polylang/polylang/pull/1404/commits/080af7116de5081d9a995ac1e1e1985d4151cfea).